### DISCUSSION TITLE
python-socketio with rabbitmq

### DISCUSSION THREAD

**Discussion by flyyabelle** [miguelgrinberg/python-socketio#827] [General]
I want to know how to send msg that from rabbitmq to client with python socketio
I have a server with python socketio, emit msg to client .the msg is from rabbitmq comsumer

**Comment by atomscaler** [comment]
The docs at python-socketio.readthedocs.io offer guidance on this:

AioPika¶
A RabbitMQ message queue is supported in asyncio applications through the AioPika package:: You need to install aio_pika with pip:

pip install aio_pika
The RabbitMQ queue is configured through the socketio.AsyncAioPikaManager class:

mgr = socketio.AsyncAioPikaManager('amqp://')
sio = socketio.AsyncServer(client_manager=mgr)

Hope this helps.

**Comment by miguelgrinberg** [comment]
@flyyabelle the integration with RabbitMQ has to be in your application. You will have to listen for messages in your RabbitMQ queue and then emit them to your Socket.IO client(s).

**Reply by flyyabelle** [reply]
can you give me a example? thanks a lot

**Reply by miguelgrinberg** [reply]
Sorry, I don't use RabbitMQ so I don't have any code to show you.
